### PR TITLE
Simplify dev-preview logging

### PR DESCRIPTION
**Pull request by mattschlebusch**
# Reason for This PR

A point raised in [PR/2976](https://github.com/firecracker-microvm/firecracker/pull/2976) is that it would be better to standardize the log level and message format for dev-preview lines,

## Description of Changes

This PR defines a dev-preview specific macro in the logger crate to standardize the logging and streamline the usage as well.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mattschlebusch** [comment]
@serban300 It isn't clear to me this revision of the dev-preview logging is better or simpler. Let me know what you think, I'm happy to discard this as well.

**Comment by serban300** [comment]
@mattschlebusch Personally I like this version better. But I would also add another parameter `msg` for example to the `log_dev_preview_warning` in order to accomodate also `warn!("{} 'load snapshot' VMM action took {} us.", DEV_PREVIEW_LOG_PREFIX, elapsed_time_us);`

**Review by serban300 — DISMISSED** [review]

**Review by luminitavoicu — COMMENTED** [review]
This looks good!
I would like this change to be introduced by only one commit, instead of 2 (one creating a macro and another one switching from macro to function). It's cleaner this way. WDYT?

File: src/vmm/src/rpc_interface.rs [hunk 1]
```diff
@@ -514,12 +514,9 @@ impl<'a> PrebootApiController<'a> {
...
     // will be replaced by a runtime controller.
     fn load_snapshot(&mut self, load_params: &LoadSnapshotParams) -> ActionResult {
-        warn!(
-            "{} {}",
-            DEV_PREVIEW_LOG_PREFIX, "Restoring snapshots is currently in development preview."
-        );
+        log_dev_preview_warning!("Virtual machine snapshots");
 
-        let load_start_us = utils::time::get_time_us(utils::time::ClockType::Monotonic);
+        let _load_start_us = utils::time::get_time_us(utils::time::ClockType::Monotonic);
```
[the remark this answers is not in the document]
Comment (mattschlebusch) [reply src/vmm/src/rpc_interface.rs:519]: You are right. This is a change made with previous logic where it was possible to be unused which failed compilation. That logic no longer applies, I can change this back.

[the remark this answers is not in the document]
Comment (mattschlebusch) [reply src/vmm/src/rpc_interface.rs:519]: > This looks good!
I would like this change to be introduced by only one commit, instead of 2 (one creating a macro and another one switching from macro to function). It's cleaner this way. You can just squash the two commits into one and modify the commit message accordingly. WDYT?

tl;dr - Sure I don't feel too strongly on this point, I will squash in a new revision.

Both commits take different approaches that I consider logically equivalent and of the same quality which is why I often don't squash commits with these circumstances.

**Review by luminitavoicu — APPROVED** [review]

**Review by serban300 — APPROVED** [review]
